stackit/internal/services/ske/cluster/datasource.go

```diff
@@ -313,7 +315,11 @@ func (r *clusterDataSource) Read(ctx context.Context, req datasource.ReadRequest
 
 func (r *clusterDataSource) getCredential(ctx context.Context, diags *diag.Diagnostics, model *Model) {
 	c := r.client
-	res, err := c.GetCredentials(ctx, model.ProjectId.ValueString(), model.Name.ValueString()).Execute()
+	res, err := c.CreateKubeconfig(ctx, model.ProjectId.ValueString(), model.Name.ValueString()).
```

Comment:
This can be simplified to:

```
payload := ske.CreateKubeconfigPayload{
		ExpirationSeconds: utils.Ptr(DefaultKubeconfigExpiration),
	}
	res, err := c.CreateKubeconfig(ctx, model.ProjectId.ValueString(), model.Name.ValueString()).CreateKubeconfigPayload(payload).Execute()
```